bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogic.java

```diff
@@ -312,7 +337,7 @@ private boolean replaceSelection(String replaceString) {
 
 		boolean replaced;
 		try {
-			replaceSelection(replaceString, isRegExSearchAvailableAndActive());
+			replaceSelection(isAvailableAndActive(SearchOptions.REGEX));
```

Comment:
why should this local state be passed anyway?

Reply:
True, I have removed that parameter and simplified and inlined this method completely in 69af3d0.